fiftyone/operators/executor.py

```diff
@@ -138,20 +138,33 @@ def execute_operator(operator_uri, ctx=None, **kwargs):
             as keyword arguments rather than including them in ``ctx``
 
     Returns:
-        an :class:`ExecutionResult`
+        an :class:`ExecutionResult`, or an ``asyncio.Task`` if you run this
+        method in a notebook context
 
     Raises:
         ExecutionError: if an error occurred while immediately executing an
             operation or scheduling a delegated operation
     """
     request_params = _parse_ctx(ctx=ctx, **kwargs)
-
-    result = asyncio.run(
-        execute_or_delegate_operator(
-            operator_uri, request_params, exhaust=True
-        )
+    coroutine = execute_or_delegate_operator(
+        operator_uri, request_params, exhaust=True
     )
-    result.raise_exceptions()
+
+    try:
+        # Some contexts like notebooks already have event loops running, so we
+        # must use the existing loop
+        loop = asyncio.get_running_loop()
+    except:
+        loop = None
+
+    if loop is not None:
+        # @todo is it possible to await result here?
+        # Sadly, run_until_complete() is not allowed in Jupyter notebooks
+        # https://nocomplexity.com/documents/jupyterlab/tip-asyncio.html
+        result = loop.create_task(coroutine)
+    else:
+        result = asyncio.run(coroutine)
+        result.raise_exceptions()
```

Comment:
Returning the coroutine is also an option, I think. Some discussion on [coroutines and tasks](https://stackoverflow.com/questions/57966935/asyncio-task-vs-coroutine)
```python
try:
    # Some contexts like notebooks already have a running event loop
    # return the coroutine for the user to await or schedule, if so
    loop = asyncio.get_running_loop()
    return coroutine
except:
    pass

result = asyncio.run(coroutine)
result.raise_exceptions()
return result
```

Reply:
Creating the task has the benefit that you don't *have* to `await` in the notebook. Empirically what I see is that the task starts immediately and everything works exactly as I would want.

If the user does need the `ExecutionResult` from the operator, then they just have to use this pattern, which isn't too bad IMO:

```py
task = foo.execute_operator(...)

# Only necessary if you want the ExecutionContext
await task
result = task.result()
```

Reply:
Makes sense 👍